packages/widget/src/utils.ts

```diff
@@ -24,3 +27,57 @@ const groupWhileHelper = <T>(
   }
   return groupWhileHelper([...currentGroup, head], tail, breakGroup);
 };
+
+// https://stackoverflow.com/a/55243880/11698825
+
+export class ReactHandler<PropsType> {
+
+  private container: any;
+  private props: any;
+
+
+  constructor(private component: any) {
```

Comment:
ts went crazy for me here and didn't want to accept react component type. We might need to update version or something